src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/DataRepository.cs

```diff
@@ -199,73 +198,71 @@ public async Task<bool> Delete(DataElement dataElement)
-            CloudBlockBlob blockBlob = cloudBlobContainer.GetBlockBlobReference(fileName);
+            BlobContainerClient cloudBlobContainer = await GetBlobContainer(org);
+            BlobClient blockBlob = cloudBlobContainer.GetBlobClient(fileName);
 
             bool result = await blockBlob.DeleteIfExistsAsync();
 
             return result;
         }
 
-        private async Task<CloudBlobContainer> GetBlobContainer(string org)
+        private async Task<BlobContainerClient> GetBlobContainer(string org)
         {
             if (_storageConfiguration.OrgPrivateBlobStorageEnabled)
             {
                 string sasToken = await _sasTokenProvider.GetSasToken(org);
-                StorageCredentials accountSasCredential = new StorageCredentials(sasToken);
 
                 string accountName = string.Format(_storageConfiguration.OrgStorageAccount, org);
                 string blobEndpoint = string.Format(_storageConfiguration.BlobEndPoint, accountName);
                 string containerName = string.Format(_storageConfiguration.OrgStorageContainer, org);
 
-                CloudStorageAccount accountWithSas = new CloudStorageAccount(accountSasCredential, new Uri(blobEndpoint), null, null, null);
-                CloudBlobClient cloudBlobClient = accountWithSas.CreateCloudBlobClient();
-                CloudBlobContainer cloudBlobContainer = cloudBlobClient.GetContainerReference(containerName);
+                StorageSharedKeyCredential storageCredentialsPrivate = new StorageSharedKeyCredential(accountName, sasToken);
+                Uri blobEndPoint = new Uri(blobEndpoint);
+                BlobContainerClient cloudBlobClient = new BlobContainerClient(blobEndPoint, storageCredentialsPrivate);
 
-                return cloudBlobContainer;
+                return cloudBlobClient;
             }
-            
-            StorageCredentials storageCredentials = new StorageCredentials(_storageConfiguration.AccountName, _storageConfiguration.AccountKey);
-            CloudStorageAccount storageAccount = new CloudStorageAccount(storageCredentials, true);
 
-            CloudBlobClient commonBlobClient = CreateBlobClient(storageCredentials, storageAccount);
-            return commonBlobClient.GetContainerReference(_storageConfiguration.StorageContainer);
+            StorageSharedKeyCredential storageCredentials = new StorageSharedKeyCredential(_storageConfiguration.AccountName, _storageConfiguration.AccountKey);
+
+            BlobServiceClient commonBlobClient = CreateBlobClient(storageCredentials, null);
+            return commonBlobClient.GetBlobContainerClient(_storageConfiguration.StorageContainer);
         }
 
-        private CloudBlobClient CreateBlobClient(StorageCredentials storageCredentials, CloudStorageAccount storageAccount)
+        private BlobServiceClient CreateBlobClient(StorageSharedKeyCredential storageCredentials, Uri blobStorageAccountUri)
```

Comment:
The parameter blobStorageAccountUri is always null. No calls provides a value. 

The else might be hit if we turn off `OrgPrivateBlobStorageEnabled` or for dev with an emulator identified with a different uri.

Reply:
Where will that URI be comming from then?

Reply:
I would assume that if `OrgPrivateBlobStorageEnabled` is false, then the other settings would match that. Maybe use the BlobEndPoint value? We should also consider removing the toggle option. Turning it off at this point would probably cause more issues than it would solve.

Reply:
Yes, i was thinking the same yesterday. Will try to remove it.

Reply:
@SandGrainOne i have now rewritten the code and removed that option